### PR TITLE
Remove unneeded references in csproj and enable GH Actions for PRs

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: src/Infrastructure/Infrastructure.csproj [hunk 1]
```diff
@@ -20,4 +18,4 @@
 	<ItemGroup>
 		<Folder Include="Data\Migrations\" />
 	</ItemGroup>
```
Comment (Youssef1313) [inline src/Infrastructure/Infrastructure.csproj:20]: This one is needed, but I can't get why? AFAIK, csproj always references *\*.cs* files, and all files in Data\Migrations\ are .cs.

Why the build fails when it's deleted

**Review by ardalis — APPROVED** [review]

File: .github/workflows/dotnetcore.yml [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-      uses: actions/setup-dotnet@v1
-      with:
-        dotnet-version: '5.0.x'
-
-    - name: Build with dotnet
-      run: dotnet build ./eShopOnWeb.sln --configuration Release
-    
-    - name: Test with dotnet
-      run: dotnet test ./eShopOnWeb.sln --configuration Release
+name: eShopOnWeb Build and Test
```
Comment (Youssef1313) [inline .github/workflows/dotnetcore.yml:1]: The change I made here was already done in https://github.com/dotnet-architecture/eShopOnWeb/pull/479. I was expecting this file to disappear after I fixed the conflicts. Not sure why it's still present in this PR without any changes.

Comment (ardalis) [reply .github/workflows/dotnetcore.yml:1]: Want to just do a clean PR with the project file fixes?

Comment (Youssef1313) [reply .github/workflows/dotnetcore.yml:1]: @ardalis Sure. Will open a new one.

**Pull request closed without merging**